scripts/zkai_attention_kv_two_head_seq32_air_private_softmax_table_lookup_gate.py

```
@@ -0,0 +1,834 @@
+    try:
+        return json.loads(raw.decode("utf-8"))
+    except (UnicodeDecodeError, json.JSONDecodeError) as err:
+        raise AttentionKvAirPrivateSoftmaxTableLookupGateError(f"failed to read {label}: {err}") from err
+
+
+def read_bounded_bytes(path: pathlib.Path, max_bytes: int, label: str) -> bytes:
+    expected_size = bounded_file_size(path, max_bytes, label)
+    try:
+        raw = path.read_bytes()
+    except OSError as err:
+        raise AttentionKvAirPrivateSoftmaxTableLookupGateError(f"failed to read {label}: {err}") from err
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)